### ISSUE TITLE
HTTP Error 503 from scribblehub.com

### ISSUE THREAD

**Issue by Ultrabenosaurus** [JimmXinu/FanFicFare#842]
I last updated stories from scribblehub.com on 9th May without issues. Today, I keep getting `HTTP Error in FFF '503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/...` both for adding new series or updating any already in my Calibre library. 

I can load all the URLs fine in a web browser, and updating from royalroad.com has no issues.

I am running Calibre 5.42 on Manjaro KDE, with the FFF plugin updated to 4.12.4. Full output from the "show job details" button in Calibre when updating a single series is below: 

![image](https://user-images.githubusercontent.com/1901532/168494120-c069b2dc-d847-440a-b2aa-8f18584355f4.png)

```
Download 1 FanFiction Book(s)
FFF: INFO: 2022-05-15 22:21:26,515: calibre_plugins.fanficfare_plugin.jobs(49): CPUs:3
FFF: INFO: 2022-05-15 22:21:26,516: calibre_plugins.fanficfare_plugin.jobs(52): FanFicFare v4.12.4
FFF: INFO: 2022-05-15 22:21:26,516: calibre_plugins.fanficfare_plugin.jobs(68): Launch background process for site www.scribblehub.com:
https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
FFF: INFO: 2022-05-15 22:21:56,346: calibre_plugins.fanficfare_plugin.jobs(130): 

================================================================================ site:(www.scribblehub.com)
FFF: INFO: 2022-05-15 22:21:27,182: calibre_plugins.fanficfare_plugin.jobs(211): 

-------------------------------------------------------------------------------- https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
FFF: DEBUG: 2022-05-15 22:21:27,190: configurable.py(1034): use_browser_cache:
FFF: DEBUG: 2022-05-15 22:21:27,190: configurable.py(1048): use_basic_cache:true
FFF: DEBUG: 2022-05-15 22:21:27,190: adapter_scribblehubcom.py(118): URL: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
FFF: DEBUG: 2022-05-15 22:21:27,190: fetcher.py(234): 
========== MISS (GET) BasicCache
https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
FFF: DEBUG: 2022-05-15 22:21:27,190: fetcher.py(469): 
---------- REQ (GET) RequestsFetcher
https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
FFF: DEBUG: 2022-05-15 22:21:56,206: fetcher.py(482): response code:503
FFF: INFO: 2022-05-15 22:21:56,207: calibre_plugins.fanficfare_plugin.jobs(439): Exception: {'title': 'Giant Robot Reincarnation?!', 'author_sort': 'pynkbites', 'author': ['pynkbites'], 'comments': '', 'good': False, 'status': 'Error', 'showerror': True, 'calibre_id': 67, 'begin': None, 'end': None, 'comment': "HTTP Error in FFF '503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/'(503)", 'url': 'https://www.scribblehub.com/series/121448/giant-robot-reincarnation/', 'site': 'www.scribblehub.com', 'series': '', 'added': False, 'pubdate': None, 'publisher': 'www.scribblehub.com', 'listorder': 0, 'collision': 'Update EPUB if New Chapters', 'uniqueurl': 'https://www.scribblehub.com/series/121448/giant-robot-reincarnation/[None-None]', 'is_adult': False, 'username': 'NoneGiven', 'password': '', 'icon': 'dialog_error.png', 'epub_for_update': '/tmp/calibre_5.42.0_tmp_7tt9mjw8/fanficfare_h22kpjhh/old-67-rmqna0ap.epub', 'calibre_columns': {}, 'outfile': '/tmp/calibre_5.42.0_tmp_7tt9mjw8/fanficfare_h22kpjhh/bgmeta-yvehywed.epub'}:HTTP Error in FFF '503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/'(503)
Traceback (most recent call last):
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 483, in request
    resp.raise_for_status() # raises RequestsHTTPError if error code.
  File "/usr/lib/python3.10/site-packages/requests/models.py", line 960, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre_plugins.fanficfare_plugin.jobs", line 260, in do_download_for_worker
    story = adapter.getStoryMetadataOnly()
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/adapters/base_adapter.py", line 308, in getStoryMetadataOnly
    self.doExtractChapterUrlsAndMetadata(get_cover=get_cover)
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/adapters/base_adapter.py", line 412, in doExtractChapterUrlsAndMetadata
    return self.extractChapterUrlsAndMetadata()
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/adapters/adapter_scribblehubcom.py", line 120, in extractChapterUrlsAndMetadata
    data = self.get_request(url)
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/requestable.py", line 114, in get_request
    return self.get_request_redirected(url,
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/requestable.py", line 106, in get_request_redirected
    (data,rurl) = self.configuration.get_fetcher().get_request_redirected(
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 398, in get_request_redirected
    fetchresp = self.do_request('GET',
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 240, in fetcher_do_request
    fetchresp = chainfn(
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 136, in fetcher_do_request
    fetchresp = chainfn(
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 371, in do_request
    fetchresp = self.request(method,url,
  File "/home/dan/.config/calibre/plugins/FanFicFare.zip/fanficfare/fetcher.py", line 504, in request
    raise exceptions.HTTPErrorFFF(
fanficfare.exceptions.HTTPErrorFFF: HTTP Error in FFF '503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/'(503)

FFF: INFO: 2022-05-15 22:21:56,346: calibre_plugins.fanficfare_plugin.jobs(142): 
Download Results:
Error https://www.scribblehub.com/series/121448/giant-robot-reincarnation/ HTTP Error in FFF '503 Server Error: Service Temporarily Unavailable for url: https://www.scribblehub.com/series/121448/giant-robot-reincarnation/'(503)

FFF: INFO: 2022-05-15 22:21:56,346: calibre_plugins.fanficfare_plugin.jobs(166): 
Error
https://www.scribblehub.com/series/121448/giant-robot-reincarnation/
```

**Comment by JimmXinu** [comment]
I'm able to download https://www.scribblehub.com/series/121448/giant-robot-reincarnation/ fine.

I would suspect either you ran into a temporary server outage(pages working in browser argues against it), or you are being blocked.  We see that happen from various sites when users download/update lots of stories/chapters.  I don't know scribblehub or your habits well enough to say how likely that is your case.  

You might also refer to [this FAQ](https://github.com/JimmXinu/FanFicFare/wiki/FAQs#why-shouldnt-i-just-select-my-entire-library-and-use-fanficfare-plugin-to-check-for-updates), since your screen capture looks like a mass update.

I'll leave this issue open for the time being, but I doubt there's anything I can do about it.

**Comment by Ultrabenosaurus** [comment]
Thanks for the reply. It makes no sense to me, but it seems like the reason was Mullvad VPN. 

Even though scribblehub loads just fine in all web browsers while Mullvad is running (connected to a server in my own country, not internationally) and I have been using Mullvad for over a year now without issues, FFF still failed to update scribblehub series when I got home from work today until I turned off Mullvad out of curiosity. 

¯\_(ツ)_/¯
